torch/_inductor/pattern_matcher.py

```diff
@@ -670,6 +667,18 @@ def run_node(self, node) -> Any:
                 else:
                     if "val" not in new.meta:
                         new.meta.update(old.meta)
+
+                    # Preserve the recompute tags in the replacement graph. We
+                    # look at the recompute tags of the original output node to
+                    # propagate the tag from the output all the way to the input
+                    # args in the replacement graph.
+                    # Note that this is best effort. Since patterns are from
+                    # many to many, there is no easy way to correctly map the
+                    # recomputable tags. It is possible in some scenrios that we
+                    # incorrectly tag some nodes as recomputables.
+                    if "recompute" in old.meta:
```

Comment:
We only want `recompute` on nodes from the forward, right. Is it worth preserving which nodes were forward and which nodes were backward in the original replace_pattern, and then using that information here ?

Reply:
We could do that. But, I don't think that will change the situation much. The outputs of replacement graph is (forward_outs, grad_out). So by moving output node by node, we are essentially percolating tags on only the forward portion of the joint graph (only forward_outs can have recompute tags set).

Reply:
This is happening after a series of other passes, which might have changed metas. It might be worth adding assertions or something that only forward_outs have recompute tags, just to sanity check